src/main/kotlin/org/pkl/intellij/util/Quadruple.kt

```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright © 2024 Apple Inc. and the Pkl project authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.pkl.intellij.util
+
+data class Quadruple<out A, out B, out C, out D>(
```

Comment:
nit: it would be better to create a specific data class even for one-shot uses. Big tuples are kind of a code smell because they are unlabeled.